src/GUI/Services/AuthService.cs

```diff
@@ -47,6 +58,16 @@ public async Task<string> Login(AuthenticationRequest request)
             }
         }
 
+        public async Task RefreshTokenAsync()
+        {
+            RefreshRequest body = new() 
+            {
+                RefreshToken = await _storage.GetItemAsync<string>(Consts.RefreshToken)
```

Comment:
Are you changing tokens every time? Mb you need check Consts.AccessTokenExpiresIn and Consts.RefreshTokenExpiresIn here?

Reply:
why is it bad?

Reply:
Because you’re doing another one request to Auth. From GUI and from service that confirm token from header.
If you would check here just time, then you can send request from GUI only when token is expired.

Reply:
I first check token, next if token expires, then i make request to auth service